### PR TITLE
Set UserAgent for ARM telemetry

### PR DESCRIPTION
**Pull request by YanaXu**
This PR is to set UserAgent for ARM request in case of SDK-based cmdlets.
And it fixes the bug of dumplicated CommandName and ParameterSetName.

### CODE REVIEW

File: src/Common/AzurePSCmdlet.cs [hunk 1]
```diff
@@ -322,6 +322,17 @@ protected virtual void SetupHttpClientPipeline()
...
             AzureSession.Instance.ClientFactory.AddUserAgent(PSVERSION, string.Format("v{0}", PowerShellVersion));
             AzureSession.Instance.ClientFactory.AddUserAgent(ModuleName, this.ModuleVersion);
+            try {
+                string hostEnv = AzurePSCmdlet.getEnvUserAgent();
+                if (!String.IsNullOrWhiteSpace(hostEnv)) 
+                {
+                    AzureSession.Instance.ClientFactory.AddUserAgent(hostEnv);
+                }
+            } 
+            catch (Exception)
```
Comment (isra-fel) [inline src/Common/AzurePSCmdlet.cs:332]: If we are try-catch `AddUserAgent()`, does it make sense to wrap all `AddUserAgent()` calls in it?

**Review by isra-fel — APPROVED** [review]
Approve with minor comments

**Review by dolauli — APPROVED** [review]

File: src/Common/AzurePSCmdlet.cs (hunk 1 above)
Comment (YanaXu) [reply src/Common/AzurePSCmdlet.cs:332]: > If we are try-catch `AddUserAgent()`, does it make sense to wrap all `AddUserAgent()` calls in it?

Here is `AzureSession.Instance.ClientFactory.AddUserAgent(hostEnv);`. Below is `ClientFactory.RemoveUserAgent(hostEnv);`. The Client part is `UserAgent += string.Format(" {0}", hostEnv);` I can't merge them together.

File: src/Common/AzurePSCmdlet.cs [hunk 2]
```diff
@@ -331,6 +342,18 @@ protected virtual void SetupHttpClientPipeline()
...
         {
+            try
+            {
+                string hostEnv = AzurePSCmdlet.getEnvUserAgent();
+                if (!String.IsNullOrWhiteSpace(hostEnv))
+                {
+                    AzureSession.Instance.ClientFactory.RemoveUserAgent(hostEnv);
+                }
+            }
+            catch (Exception)
```
Comment (YanaXu) [inline src/Common/AzurePSCmdlet.cs:353]: I add a catch here just in case the env var is not OK for setting UserAgent, we'll ignore it.

**Pull request merged**